### PR TITLE
allow us to disable unattended upgrades for ubuntu

### PR DESCRIPTION
**Pull request by paulgmiller**
**What type of PR is this?**

/kind feature
**What this PR does / why we need it**:
Unattended upgrade can be harmful and redundant if you have node image uprgade. 

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: pkg/agent/baker.go [hunk 1]
```diff
@@ -316,6 +316,9 @@ func getContainerServiceFuncMap(config *datamodel.NodeBootstrappingConfiguration
 		"Disable1804SystemdResolved": func() bool {
 			return config.Disable1804SystemdResolved
 		},
+		"EnableUnattendedUpgrade": func() bool {
+			return !config.DisableUnattendedUpgrades
```
Comment (yewmsft) [inline pkg/agent/baker.go:320]: nit: shall we call it enable to avoid the double negative?

Comment (paulgmiller) [reply pkg/agent/baker.go:320]: default for a bool is false though and for now the default needs to be unattended upgrades enabled.

File: parts/linux/cloud-init/artifacts/cse_main.sh [hunk 2]
```diff
@@ -227,7 +227,9 @@ if $FULL_INSTALL_REQUIRED; then
 fi
 
 {{- /* re-enable unattended upgrades */}}
+{{- if EnableUnattendedUpgrade }} 
 rm -f /etc/apt/apt.conf.d/99periodic
```
Comment (alexeldeib) [inline parts/linux/cloud-init/artifacts/cse_main.sh:231]: how about we remove this, and then in cloud init, we set the correct value (enabled/disabled) based on the bool?

**Review by alexeldeib — APPROVED** [review]
lgtm. long term i'd prefer disabling in the base image and consume user data as an enable/disable mechanism but that's probably not happening soon. 

actually would be nice to disable in the base image and enable when required just to avoid races...but still fine either way

File: pkg/agent/baker.go (hunk 1 above)
Comment (paulgmiller) [reply pkg/agent/baker.go]: with new refactor at least go rid of the not

**Pull request merged**